server/src/test/java/server/haengdong/domain/BillActionTest.java

```diff
@@ -0,0 +1,56 @@
+package server.haengdong.domain;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class BillActionTest {
+
+    @DisplayName("앞뒤 공백을 제거한 title이 2 ~ 30자가 아니면 지출을 생성할 수 없다.")
+    @ParameterizedTest
+    @ValueSource(strings = {" 감 ", "", " ", "감자감자감자감자감자감자백호백호백호백호백호감자감자감자감자감자감자백호백호백호백호백호"})
```

Comment:
```suggestion
    @ValueSource(strings = {" 감 ", "", " ", "1234567890123456789012345678901"})
```
c: 엣지 케이스인 정확한 31자를 검증하면 좋겠습니다. 저는 길이 제한을 검증할 때 31자라는 것을 쉽게 인지할 수 있는 방식을 선호합니다.